attachment.go

```diff
@@ -4,14 +4,21 @@ import (
 	"bytes"
 	"context"
 	"fmt"
-	"io"
-
 	"github.com/ProtonMail/gopenpgp/v2/crypto"
 	"github.com/go-resty/resty/v2"
+	"io"
 )
 
 func (c *Client) GetAttachment(ctx context.Context, attachmentID string) ([]byte, error) {
-	return c.attPool().ProcessOne(ctx, attachmentID)
+	buffer := bytes.NewBuffer(nil)
```

Comment:
It's actually more standard in Go to use `new(bytes.Buffer)`, might be useful to avoid an unnecessary function call here (if Go doesn't inline it). The stdlib docstring confirms:

```
// NewBuffer creates and initializes a new Buffer using buf as its
// initial contents. The new Buffer takes ownership of buf, and the
// caller should not use buf after this call. NewBuffer is intended to
// prepare a Buffer to read existing data. It can also be used to set
// the initial size of the internal buffer for writing. To do that,
// buf should have the desired capacity but a length of zero.
//
// In most cases, new(Buffer) (or just declaring a Buffer variable) is
// sufficient to initialize a Buffer.
```

Reply:
Replaced with version that does not allocate the buffer in memory.